### ISSUE TITLE
Owl Carousel Item Class not recognized

### ISSUE THREAD

**Issue by LukeDouglas2** [gantry/gantry5#1947]
I have modified the owlcarousel.yaml file to add an 'itemclass' as follows:

```
 .buttonclass:
          type: input.text
          label: Button Class
          description: Input the button class.
          default: 'button-outline'
        .itemclass:
          type: input.text
          label: Item Class
          description: Input the item class.
          default: 'item-outline'
```
I have modified the owlcarousel.html.twig file to add the new 'itemclass' field as follows (noticed the 'z' in front and behind the code to insert the item class field (cloned / modified from the button class):

```

<div id="g-owlcarousel-{{ id }}" class="g-owlcarousel owl-carousel {% if particle.imageOverlay == 'enable' %}has-color-overlay{% endif %}">
            {% for item in particle.items %}
                <div class="g-owlcarousel-item owl-item z{{ item.itemclass|e }}z">
                    <div class="g-owlcarousel-item-wrapper">
                        <div class="g-owlcarousel-item-img">
                            <img src="{{ url(item.image)|e }}" alt="{{ item.title|e }}" />
                        </div>
                        <div class="g-owlcarousel-item-content-container">
                            <div class="g-owlcarousel-item-content-wrapper">
                                <div class="g-owlcarousel-item-content">
                                    {% if item.title %}
                                        <h1 class="g-owlcarousel-item-title">{{ item.title|raw }}</h1>{% endif %}
                                    {% if item.desc %}
                                        <h2 class="g-owlcarousel-item-desc">{{ item.desc|raw }}</h2>{% endif %}
                                    {% if item.link %}
                                        <div class="g-owlcarousel-item-link">
                                            <a target="_self" class="g-owlcarousel-item-button button {{ item.buttonclass|e }}" href="{{ item.link|e }}">
                                                {{ item.linktext|raw }}
                                            </a>
                                        </div>
                                    {% endif %}
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            {% endfor %}

        </div>
```

Here is the rendered HTML code - noticed the 'zz' class which is the beginning 'z' and the ending 'z' so I know it's recognizing my code but just not inserting the field content.   

```
<div class="g-owlcarousel-item owl-item zz">
                    <div class="g-owlcarousel-item-wrapper">
                        <div class="g-owlcarousel-item-img">
                            <img src="/newdesign/images/banners/bg-author-1800x1280.jpg?58dd7caf" alt="&amp;nbsp;">
                        </div>
                        <div class="g-owlcarousel-item-content-container">
                            <div class="g-owlcarousel-item-content-wrapper">
                                <div class="g-owlcarousel-item-content">
                                                                            <h1 class="g-owlcarousel-item-title">&nbsp;</h1>                                                                            <h2 class="g-owlcarousel-item-desc"><span class="owlcarousel-name width100 text-left fontsize-150 w3-animate-right">Author</span><br><span class="owlcarousel-subtitle width100 text-left w3-animate-left"><span class="text-uppercase">Young Adult - Women’s Literature - Non-Fiction</span><br><small> Escape the day to day with a space trip, some humorous romance, or a dose of comically real life.</small></span></h2>                                                                            <div class="g-owlcarousel-item-link">
                                            <a target="_self" class="g-owlcarousel-item-button button button-outline" href="https://kathryndfuller.com/newdesign/index.php?option=com_content&amp;view=article&amp;id=27">
                                                Read More
                                            </a>
                                        </div>
                                                                    </div>
                            </div>
                        </div>
                    </div>
                </div>
```

Can you tell me what I need to do in order to assign a class to each individual Owl Carousel item?

Thanks.

Luke

**Comment by newkind** [comment]
Make sure that indentation in YAML is proper, re-save all OwlCarousel items, clear cache.

**Comment by LukeDouglas2** [comment]
That was my first concern which I have doubled checked both in HTML-KIT and in the cPanel editor.  No tabs, just spaces and they match up exactly to the button class which I simply copied/pasted, then edited.  I wish I could give you a URL but the site is in development and offline. :(

**Comment by newkind** [comment]
If this is a particle you should have it's configuration stored in the `custom` folder. Can you please check the yaml file to see if it contains your custom css class?

**Comment by LukeDouglas2** [comment]
When I duplicated the default page, it created the folder 'templates\g5_helium\custom\config\14'.  The edit I made to the owlcarousel.yaml file for the new class is in that folder.

I have a custom.scss file in the 'templates\g5_helium\custom\scss' folder which is where i have the custom.scss file located where I do all my styling.

The owlcarousel.html.twig and owlcarousel.yaml files that I edited and which are listed above are in the "templates/g5_helium/particles" folder.

Is this correct?

**Comment by newkind** [comment]
Are you sure that you're editing proper outline?
Why are you modifying the yaml manually and not the Owl in admin?

**Comment by LukeDouglas2** [comment]
Newkind,  you cannot add new Owl Carousel fields from within the Owl admin.  If you can, I wish you would share that knowledge with me.  It sounds like you don't understand the issue I am having.

The issue is that the individual Owl Carousel 'individual' items do NOT have a class field (similar to the button class field) and I need the ability to do unique styling for each item.  I really can't use the 'nth child selector' option as the items will be changing from month to month which will change the 'nth child' location.

**Comment by newkind** [comment]
I'm not talking about "adding fields" in the admin - that's not possible. You can only add new fields in the YAML. If you edited the YAML properly your new field should appear in the OwlCarousel settings. That's what I was referring to - editing your new input in the admin.

**Comment by LukeDouglas2** [comment]
newkind,

The new fields DOES appear in the OwlCarousel settings.  Read my original post again.  

The issue is that the 'class' field is NOT being shown in the webpage.  I know this for a 'fact' as I put a 'z' in front of and behind the 'class' field in the TWIG file [ <div class="g-owlcarousel-item owl-item z{{ item.itemclass|e }}z"> ].  When the webpage is 'rendered', there is a 'zz' class so it is picking up the new TWIG code but it's just not inserting the 'item.itemclass' field.  

Again, I used the same formatting as the button class.  Either I am not formatting this properly or I am missing a step.

**Comment by newkind** [comment]
So you added your custom class `item-outline` in the admin or manually you typed it in the saved configuration of OwlCarousel?

Are you sure that you edited proper outline ? I don't see a reason why it wouldn't work. If you want to you can attach your twig+yaml file and I'll test them on my local.

**Comment by LukeDouglas2** [comment]
Newkind,

I've attached the zip file for the 'custom/config/14' folder and the two files from the 'particles' folder.

[14.zip](https://github.com/gantry/gantry5/files/891209/14.zip)
[particles.zip](https://github.com/gantry/gantry5/files/891210/particles.zip)

**Comment by newkind** [comment]
Two things :

1) You named your field `.owlitemclass` but in your twig file you name it `{{ item.itemclass|e }}` where it should  be `{{ item.owlitemclass|e }}`

2) I don't see any reference to `owlitemclass` in the `layout.yaml` file so are you sure that you changed settings in proper outline/particle?

![Image]https://monosnap.com/file/rQWSPAiufkllZuhFCWgi2nUMjNCjP8.png

**Comment by LukeDouglas2** [comment]
newkind,

Thank you!  I knew another pair of eyes could find the issue.  I changed the 'item.owlitemclass' to 'item.itemclass' and it works!  

Now of all things I had an update for the Helium template which overwrote the changes I made to the owlcarousel.yaml and owcarousel.html.twig files.  I cross-referenced both of the new files to the original files and there were not changes. Luckily, I had created a backup of my edit changes so I overwrote the files with my changes.  I did create a 'custom/particles' folder and put the two files into that folder while leaving the originals in the 'h5_helium/particles' folder.  It seems to work just fine.  Is this the correct procedure to ensure my edits are not overwritten by future template updates?
